Makefile

```diff
@@ -57,17 +65,37 @@ run-local: install-apis ## Install CRDs and run provisioner locally
 	kubectl create namespace rukpak-system
 	$(Q)go run provisioner/k8s/main.go
 
+deploy: install-apis ## Deploy the operator to the current cluster
+	kubectl apply -f provisioner/k8s/manifests
+
+run: build-local-container load-image deploy
+
+## --------------------------------------
+## Build and Load
+## --------------------------------------
+
 # Binary builds
-GO_BUILD := $(Q)go build
+.PHONY: build bin/k8s bin/unpack
 
-build: bin/k8s
+GO_BUILD := $(Q)go build
+build: bin/k8s bin/unpack
 
 bin/k8s:
 	CGO_ENABLED=0 go build -o $@ ./provisioner/k8s
 
 bin/unpack:
 	CGO_ENABLED=0 go build -o $@ ./cmd/unpack/...
 
+build-local-container: ## Builds provisioner container image locally
+	docker build -f Dockerfile -t $(IMAGE) .
+
+kind-load: ## Load-image loads the currently constructed image onto the cluster
+	 ${KIND} load docker-image $(IMAGE)
+
+kind-cluster: ## Standup a kind cluster for e2e testing usage
+	${KIND} get clusters | grep ${KIND_CLUSTER_NAME} || ${KIND} create cluster --name ${KIND_CLUSTER_NAME}
+
+e2e-kind: build-local-container kind-cluster kind-load deploy test-e2e ## Run e2e tests against a kind cluster
```

Comment:
WDYT about being more opinionated and setting this up such that `test-e2e` _always_ uses kind.

IMO, hermtically sealing as much as possible is and leaving fewer choices to developers means more guarantees of consistency between local and CI test runs.

Reply:
Yeah I don't know where I land with this most of the time. I think the idea of hermetically sealing clusters makes a lot of sense, but the potential churn of create a cluster, waiting for that cluster to be ready, installing the rukpak stack, running e2e tests, and then deleting the cluster isn't the greatest experience when you're developing e2e tests and just want quick feedback from the cluster.

I think this is still worth pursuing, but maybe it's fine in the short term to have separate Makefile targets that enable both workflows while we iterate? We could track hermetically sealing cluster runs (or even make the OLM kind provisioner a library we import here) as an issue.

Reply:
Yeah, my thought would be something like:
```
# Set up the environment with a new kind cluster
# Run each of the actual e2e tests
# Tear down the kind cluster
# Restore the original environment
make e2e  

# Just run the e2e against the current environment
make e2e-provisioner-k8s 
```

I guess that's really close to what you've got, and maybe all I'm suggesting is this, such that a "default"/"obvious" target is the hermetically sealed one, but sub-targets exist if you want to manage setup/teardown yourself:
```suggestion
e2e: build-local-container kind-cluster kind-load deploy test-e2e ## Run e2e tests against a kind cluster
```

Reply:
I took in your suggested change in the last force push. Maybe we can experiment with that `kind-cluster` target and switch the conditional to check whether there's an _existing_ cluster with that special e2e kind cluster name, and if so, delete that cluster and create a new one? 

Existing:

```make
kind-cluster: ## Standup a kind cluster for e2e testing usage
	${KIND} get clusters | grep ${KIND_CLUSTER_NAME} || ${KIND} create cluster --name ${KIND_CLUSTER_NAME}
```

And then we could update that conditional to something like the following:

```make
kind-cluster: ## Standup a kind cluster for e2e testing usage
ifeq (1, $(shell kind get clusters | grep ${KIND_CLUSTER_NAME} | wc -l))
	${KIND} delete cluster --name ${KIND_CLUSTER_NAME}
endif
        ${KIND} create cluster --name ${KIND_CLUSTER_NAME}
```

Reply:
I tackled partially implementing this in https://github.com/operator-framework/rukpak/pull/90. Should leave a more consistent cluster cluster state between local and CI runs.